### DISCUSSION TITLE
How to pass custom data- attributes as props to child component when using HTML template literals

### DISCUSSION THREAD

**Discussion by chvelkov** [solidjs/solid#1829] [Q&A]
Hello,
I'm using HTML template literals to define some components and need to pass some custom data- attributes. Using JSX, we could do:

```
function InnerComp(props) {
  return <div {...props}>Inner</div>
}

function CompJSX() {
  return <InnerComp data-test="2"></InnerComp>;
}
```

However this doesn't work with CompHTML, which uses template literals - the template won't compile, because dashes "-" are not allowed in variable names:

```
function CompHTML() {
  return html`<${InnerComp} data-test="2"><//>`;
}
```

I didn't find anything in the documentation about how to handle such cases. How would you pass "data-test" in this case?
One possibility is of course to manually parse the props in `InnerComp` and convert camelCase `dataTest` to `data-test`, but I was wondering whether there's a better solution.

Here's a playground link:

https://playground.solidjs.com/anonymous/46be69a2-71a2-4ea8-abf6-29329ec6a160

**Comment by atk** [comment]
This is using lit under the hood, so you should be able to use `.dataTest=...` to assign the property (just add a leading dot).

**Reply by chvelkov** [reply]
Hm, I just tried it in the playground link, but it doesn't compile. I get `Uncaught SyntaxError: Unexpected token '.'`

**Reply by atk** [reply]
Sorry, mixed up lit-dom-expressions with lit. You can use Datalist: https://playground.solidjs.com/anonymous/d5970b60-9256-4462-ae05-3c58f8b78484

**Reply by chvelkov** [reply]
Thank you! However, this isn't working as expected - it results in `<div datalist="[object Object]">Inner</div>` while the JSX correctly renders `<div data-test="2">Inner</div>`.

If it isn't possible to set data- attributes currently, I guess I'll have to resort to passing `data_test` as property key and doing a string replace of all underscores with dashes.
